### PR TITLE
Introduce sieves

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: tests/python/pipeline/sieves/test_os_sieve.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            })
+        )
+        os_sieve = OperatingSystemSieve(graph=None, project=project)
+        os_sieve.run(sieve_context)
+
+        expected = {
+            ("pytest", "3.0.0", "https://pypi.org/simple"),
+            ("tensorflow", "1.9.0", "https://pypi.org/simple"),
+            # Filtering out this entry is left on another sieve which ensures runtime environment compatibility.
+            ("tensorflow", "1.9.0", "https://tensorflow.pypi.thoth-station.ninja/index/manylinux2010/jemalloc/simple/"),
```
Comment (harshad16) [inline tests/python/pipeline/sieves/test_os_sieve.py]: we don't have tensorflow==1.9.0 for manylinux2010 in our index yet.

Comment (fridex) [reply tests/python/pipeline/sieves/test_os_sieve.py]: I just put some value - these values are artificial for test cases, not prod code.

File: tests/python/pipeline/sieves/test_os_sieve.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+        assert set(sieve_context.iter_direct_dependencies_tuple()) == expected
+
+    def test_os_sieve_no_remove(self):
+        """Test the TensorFlow package is not removed as it has no other candidate."""
+        package_versions = [
+            PackageVersion(
+                name="tensorflow",
+                version="==1.9.0",
+                index=Source("https://tensorflow.pypi.thoth-station.ninja/index/fedora/30/jemalloc/simple/"),
```
Comment (harshad16) [inline tests/python/pipeline/sieves/test_os_sieve.py]: we don't have fedora30 tensorflow in our index as well. Also the index hasn't been changed in this format:
_https://tensorflow.pypi.thoth-station.ninja/index/**fedora/30**/jemalloc/simple/_

Comment (fridex) [reply tests/python/pipeline/sieves/test_os_sieve.py]: Yes, that's why I commented out this implementation for now.

File: tests/python/pipeline/sieves/test_os_sieve.py (hunk 1 above)
Comment (harshad16) [reply tests/python/pipeline/sieves/test_os_sieve.py]: ah, thanks for the clarification.

**Comment by harshad16** [comment]
LGTM :tada: 
PTAL @goern  :smile:

**Comment by fridex** [comment]
> LGTM 
> PTAL @goern

I will merge this as my next work depends on this change. If you have any comments, please state them and I will adjust sources accordingly.

**Pull request merged**